src/main/kotlin/org/ozwillo/dcimporter/web/MarchePublicHandler.kt

```diff
@@ -477,6 +488,284 @@ class MarchePublicHandler(private val datacoreProperties: DatacoreProperties,
+
+                        }
+                        else -> registreType.toDcObject(datacoreProperties.baseUri, registreType.cle!!)
+                    }
+
+                    try {
+                        datacoreService.getResourceFromIRI(MP_PROJECT, type, "FR/$siret/${req.pathVariable("reference")}/${registreType.cle}", bearer)
+                        datacoreService.updateResource(MP_PROJECT, type, dcRegistre, bearer)
+                    }catch (e: HttpClientErrorException){
+                        when(e.statusCode){
+                            HttpStatus.NOT_FOUND -> datacoreService.saveResource(MP_PROJECT, type, dcRegistre, bearer)
+                            else -> throw e
+                        }
+                    }
+                }
+                .collectList()
+                .flatMap {
+                    ok().contentType(MediaType.APPLICATION_JSON).body(BodyInserters.empty<String>())
+                }
+                .onErrorResume(this::throwableToResponse)
+    }
+
+    fun getRegistreForConsultation(req: ServerRequest): Mono<ServerResponse>{
+        val bearer = extractBearer(req.headers())
+        val siret = req.pathVariable("siret")
+        val iri = "FR/$siret/${req.pathVariable("reference")}"
+        val type = req.pathVariable("type")
+        val subject = when(type){
+            MSUtils.REPONSE_TYPE -> "mpreponse:consultation"
+            MSUtils.RETRAIT_TYPE -> "mpretrait:consultation"
+            else -> throw HttpClientErrorException(HttpStatus.BAD_REQUEST, "Unable to recognize register type from request : reponse_0 or retrait_0")
+        }
+
+        val startParam = if (req.queryParam("start").isPresent) req.queryParam("start").get().toInt() else 0
+        val maxParam = if (req.queryParam("limit").isPresent) req.queryParam("limit").get().toInt() else 50
+
+        try {
+            datacoreService.getResourceFromIRI(MP_PROJECT, ORG_TYPE, "FR/$siret", bearer)
+        } catch (e: HttpClientErrorException) {
+            val body = when(e.statusCode) {
+                HttpStatus.UNAUTHORIZED -> "Token unauthorized, maybe it is expired ?"
+                HttpStatus.NOT_FOUND -> "Organization with SIRET $siret does not exist"
+                else -> "Unexpected error"
+            }
+
+            return status(e.statusCode).body(BodyInserters.fromObject(body))
+        }
+
+        return try {
+            val body: MutableList<Registre> = arrayListOf()
+
+            val dcConsultation = datacoreService.getResourceFromIRI(MP_PROJECT, CONSULTATION_TYPE, iri, bearer)
+            datacoreService.findResources(MP_PROJECT, type, DCQueryParameters(subject, DCOperator.EQ, DCOrdering.DESCENDING, dcConsultation.getUri()), startParam, maxParam)
+                    .collectList()
+                    .flatMap { registres ->
+                        registres.forEach {registre ->
```

Comment:
it can be improved without the forEach (and the mutable body)

Reply:
Ok I updated it